File: mobile-android/app/src/main/java/com/nexusops/viewmodel/DashboardViewModel.kt
package com.nexusops.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.nexusops.data.NexusRepository
import com.nexusops.model.WorkflowItem
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

data class DashboardUiState(
    val workflows: List<WorkflowItem> = emptyList(),
    val isLoading: Boolean = false,
    val systemHealthScore: Double = 99.4,
    val activeWorkflowsCount: Int = 14
)

class DashboardViewModel(private val repository: NexusRepository = NexusRepository()) : ViewModel() {
    private val _uiState = MutableStateFlow(DashboardUiState())
    val uiState: StateFlow<DashboardUiState> = _uiState.asStateFlow()

    init {
        loadDashboardData()
    }

    fun loadDashboardData() {
        viewModelScope.launch {
            _uiState.value = _uiState.value.copy(isLoading = true)
            val workflows = repository.getWorkflows()
            _uiState.value = _uiState.value.copy(
                workflows = workflows,
                isLoading = false
            )
        }
    }
}

File: mobile-ios/ViewModels/DashboardViewModel.swift
import Foundation
import Combine

@MainActor
class DashboardViewModel: ObservableObject {
    @Published var workflows: [WorkflowItem] = []
    @Published var isLoading: Bool = false
    @Published var systemHealthScore: Double = 99.4
    @Published var activeWorkflowsCount: Int = 14

    func loadDashboardData() async {
        isLoading = true
        defer { isLoading = false }
        
        do {
            self.workflows = try await NetworkService.shared.fetchWorkflows()
        } catch {
            print("Error loading dashboard data: \(error)")
        }
    }
}
